concordium-consensus/src/Concordium/GlobalState/TreeState.hs

```diff
@@ -85,6 +85,20 @@ data AddTransactionResult =
   NotAdded !TVer.VerificationResult
   deriving(Eq, Show)
 
+-- |Status of a "recent" block. Recent here means that instead of looking up the
+-- full status of a block that is older than the last finalized one, we return
+-- only the fact that it is older than last finalized. This performs better in
+-- cases where this is the only information that is needed. It avoids loading
+-- finalized blocks from the database.
+data RecentBlockStatus bp pb  =
+  -- |The block is either pending, dead, or no older than the last finalized block.
+  RecentBlock !(BlockStatus bp pb)
+  -- |The block is known, and is strictly older than the last finalized block.
+  | OlderThanLastFinalized
```

Comment:
I think a better name like `OldFinalized` would be more appropriate, because this indicates that the block is finalized, which I think should be emphasised. (A dead block can fit the description "strictly older than the last finalized block".)

Reply:
Ok.